tests/add_operator_labels/add_operator_labels_suite_test.go

```diff
@@ -0,0 +1,42 @@
+package add_operator_labels
+
+import (
+	"github.com/k8ssandra/cass-operator/pkg/oplabels"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+/**
+A valid label must be an empty string or consist of alphanumeric characters,
+'-', '_' or '.', and must start and end with an alphanumeric.
+*/
+func TestLabelNameClean(t *testing.T) {
```

Comment:
tests directory is for e2e tests, unit tests should be in the pkg/labels. Also, could we have additionally some real ones here? Like

"TestCluster" should come out as "TestCluster", while "Test Cluster" should be without the whitelabel and so on. It makes it easier for someone to verify their result if it's not just theoretical instances here. 

And also we need to verify that the values that we currently accept (and that do work) are not changed. Otherwise all existing clusters would stop working.

Reply:
> And also we need to verify that the values that we currently accept (and that do work) are not changed. Otherwise all existing clusters would stop working.

Do we have this list of accepted values, I'm not familiar with the set being used?

Reply:
The accepted values are all values that someone could have used. Cassandra itself limits everything that can be cast to UTF-16. So all values that are accepted by regexp `(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?` should be valid moving forward.